### PR TITLE
Optimize Dockerfiles to remove the redundant steps in the release script

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

### What does this PR do?
Adds a couple of minor optimizations to the Dockerfiles in order to get rid of the redundant steps in the existing release script and to avoid it in the new one I'm currently working on.

### CODE REVIEW

File: dockerfiles/theia/Dockerfile [hunk 1]
```diff
@@ -20,9 +20,6 @@ ENV GITHUB_TOKEN=$GITHUB_TOKEN
 
 ARG THEIA_GITHUB_REPO=eclipse-theia/theia
 
-# Define upstream version of theia to use
-ARG THEIA_VERSION=master
```
Comment (azatsarynnyy) [inline dockerfiles/theia/Dockerfile:24]: as I understood, it has been reworked to come from the `build.include`
https://github.com/eclipse/che-theia/blob/287faa2566b8d7b9da8c335c580d8956054dfe22/build.include#L14
and looks like it's not used anymore

Comment (azatsarynnyy) [reply dockerfiles/theia/Dockerfile:24]: @benoitf correct me please if I'm wrong here

File: dockerfiles/theia-dev/e2e/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ARG GITHUB_TOKEN=''
 ENV GITHUB_TOKEN=$GITHUB_TOKEN
 
 # Just try to build the latest theia with current image
-RUN git clone -b 'master' --single-branch --depth 1 https://github.com/eclipse-theia/theia theia
+RUN git clone -b ${GIT_BRANCH_NAME} --single-branch --depth 1 https://github.com/eclipse-theia/theia theia
```
Comment (azatsarynnyy) [inline dockerfiles/theia-dev/e2e/Dockerfile:16]: it's odd... build is failed on CI but Docker build is passed on my side with the correct value substituted
I'll figure it out tomorrow

File: dockerfiles/theia/Dockerfile (hunk 1 above)
Comment (benoitf) [reply dockerfiles/theia/Dockerfile:24]: https://github.com/eclipse/che-theia/blob/master/dockerfiles/theia/docker/alpine/builder-patch-theia.dockerfile#L5-L7 ?

**Comment by azatsarynnyy** [comment]
it's outdated as a new release script has been introduced in #709

**Pull request closed without merging**